tests/phpunit/tests/theme/globalStylesheet.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+/**
+ * @group themes
+ */
+class Tests_Global_Stylesheet extends WP_UnitTestCase {
+
+	public function test_block_theme_using_variables() {
```

Comment:
Should we try to incorporate this logic:
```
	public function set_up() {
		global $wp_theme_directories;

		parent::set_up();

		$backup_wp_theme_directories = $wp_theme_directories;
		$wp_theme_directories        = array( WP_CONTENT_DIR . '/themes' );

		add_filter( 'extra_theme_headers', array( $this, 'theme_data_extra_headers' ) );
		wp_clean_themes_cache();
		unset( $GLOBALS['wp_themes'] );
	}

	public function tear_down() {
		global $wp_theme_directories;

		$wp_theme_directories = $this->wp_theme_directories;

		remove_filter( 'extra_theme_headers', array( $this, 'theme_data_extra_headers' ) );
		wp_clean_themes_cache();
		unset( $GLOBALS['wp_themes'] );

		parent::tear_down();
	}
```

As in tests/phpunit/tests/theme.php


It may be the reason $wp_theme_directories is not countable and some tests fail.

Reply:
Nice lead, thanks! I ended up doing https://github.com/WordPress/wordpress-develop/pull/2233/commits/8647bf4901f5472246482aa50e19808c73456ae6 The `wp_theme_directories` global was empty, the theme dir wasn't set up. This should make things more reproducible/stable for these tests.